prov/gni/src/gnix_ep.c

```diff
@@ -429,18 +432,19 @@ int gnix_ep_open(struct fid_domain *domain, struct fi_info *info,
 	 */
 
 	if (nic) {
-		gnix_list_del_init(&nic->list);
-		list_add_tail(&gnix_nic_list, &nic->list);
+		gnix_list_del_init(&nic->gnix_nic_list);
+		list_add_tail(&gnix_nic_list, &nic->gnix_nic_list);
 	}
 
 	/*
 	 * no nic found create a cdm and attach
 	 */
 	if (!nic) {
```

Comment:
For future reference, if you decide to refactor this part it would look cleaner and make a bit more sense as something like:
```
if (nic) {
  // nic found, balance use by removing from head and positioning at end
  gnix_list_del_init(&nic->gnix_nic_list);
} else {
  nic = create_cdm(...);
}
list_add_tail(&gnix_nic_list, &nic->gnix_nic_list);
```